### PR TITLE
rtos: io_drivers: add loopback mode description

### PR DESCRIPTION
**Pull request by marcinszkudlinski**
This commit adds a description of using SSP devices in loopback mode

### CODE REVIEW

File: architectures/firmware/sof-zephyr/rtos_layer/io_drivers/i2s/i2s_driver.rst [hunk 1]
```diff
@@ -108,3 +108,15 @@ stream with already running ones.
 
 "Single" I2S links may be synchronized and aggregated by sending I2sSyncData to
 the I2S IO Driver.
+
+Loopback mode
+======================================
+
+In some architectures I2S transmitter and receiver share same physical data pins - 
+the Tx pin of the transmitter is connected to the Rx pin of the receiver.
```
Comment (plbossart) [inline architectures/firmware/sof-zephyr/rtos_layer/io_drivers/i2s/i2s_driver.rst:116]: that's not quite right. The connection is at the IP level, not at the physical data pins. We can enable the loopback even if the pins are not connected to anything, e.g. on an Up board or RVP.

**Review by plbossart — APPROVED** [review]

**Review by mwasko — DISMISSED** [review]

**Review by plbossart — DISMISSED** [review]

File: architectures/firmware/sof-zephyr/rtos_layer/io_drivers/i2s/i2s_driver.rst [hunk 2]
```diff
@@ -108,3 +108,18 @@ stream with already running ones.
 
 "Single" I2S links may be synchronized and aggregated by sending I2sSyncData to
 the I2S IO Driver.
+
+Loopback mode
+======================================
+
+I2S transmitter and receiver share data pins at IP level - the Tx pin of the transmitter
```
Comment (deb-intel) [inline architectures/firmware/sof-zephyr/rtos_layer/io_drivers/i2s/i2s_driver.rst:115]:
```suggestion
The I2S transmitter and receiver share data pins at the IP level. The Tx pin of the transmitter 
```

File: architectures/firmware/sof-zephyr/rtos_layer/io_drivers/i2s/i2s_driver.rst [hunk 3]
```diff
@@ -108,3 +108,18 @@ stream with already running ones.
...
 the I2S IO Driver.
+
+Loopback mode
+======================================
+
+I2S transmitter and receiver share data pins at IP level - the Tx pin of the transmitter 
+is connected to the Rx pin of the receiver. This may be used for easy creation of digital 
+loopbacks (LBM, loopback mode) - the receiver always does see what the sender is sending. 
+
+Please note that
```
Comment (deb-intel) [inline architectures/firmware/sof-zephyr/rtos_layer/io_drivers/i2s/i2s_driver.rst:119]:
```suggestion
Please note the following: 
```

File: architectures/firmware/sof-zephyr/rtos_layer/io_drivers/i2s/i2s_driver.rst [hunk 4]
```diff
@@ -108,3 +108,18 @@ stream with already running ones.
...
+Loopback mode
+======================================
+
+I2S transmitter and receiver share data pins at IP level - the Tx pin of the transmitter 
+is connected to the Rx pin of the receiver. This may be used for easy creation of digital 
+loopbacks (LBM, loopback mode) - the receiver always does see what the sender is sending. 
+
+Please note that 
+
+ - all the parameters of transmitter and receiver, like number of channels, data rates
```
Comment (deb-intel) [inline architectures/firmware/sof-zephyr/rtos_layer/io_drivers/i2s/i2s_driver.rst:121]:
```suggestion
 - all the parameters of transmitter and receiver, like number of channels, data rates, and format must match each other
```

File: architectures/firmware/sof-zephyr/rtos_layer/io_drivers/i2s/i2s_driver.rst [hunk 5]
```diff
@@ -108,3 +108,18 @@ stream with already running ones.
...
+======================================
+
+I2S transmitter and receiver share data pins at IP level - the Tx pin of the transmitter 
+is connected to the Rx pin of the receiver. This may be used for easy creation of digital 
+loopbacks (LBM, loopback mode) - the receiver always does see what the sender is sending. 
+
+Please note that 
+
+ - all the parameters of transmitter and receiver, like number of channels, data rates 
+and format, must match each other
```
Comment (deb-intel) [inline architectures/firmware/sof-zephyr/rtos_layer/io_drivers/i2s/i2s_driver.rst:122]:
```suggestion

```

**Review by deb-intel — CHANGES_REQUESTED** [review]
Edits made to improve style and readability.

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
Applied @deb-intel changes, pending CI.

**Pull request merged**